lib/ansible/modules/source_control/gitlab_project_variable.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, Markus Bergholz (markuman@gmail.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+module: gitlab_project_variable
+short_description: Creates/updates/deletes Gitlab Projects Variables
+description:
+  - When a project variable does not exists, it will be created.
+  - When a project variable does exists, its value will be updated when the values are different.
+  - Variables which are presented in the playbook, but are not presented in the Gitlab project,
+    they stay untouched (purged_vars = False) or will be deleted (purged_vars = True).
+version_added: "2.9"
+author:
+    - "Markus Bergholz (@markuman)"
+requirements:
+  - python >= 2.7
+  - python-gitlab python module
+options:
+  state:
+    description:
+      - create or delete project variable.
+      - Possible values are present and absent.
+    default: present
+    type: str
+    choices: ["present", "absent"]
+  server_url:
+    description:
+      - The URL of the Gitlab server, with protocol (i.e. http or https).
+    required: true
+    type: str
+  login_token:
+    description:
+      - Gitlab access token with api permissions.
+    required: true
+    type: str
+  name:
+    description:
+      - The path and name of the project
+    required: true
+    type: str
+  purge_vars:
+    description:
+      - When set to true, all variables which are not presented in the task will be deleted.
+    default: false
+    required: false
+    type: bool
+  vars:
+    description:
+      - A list of key value pairs
+    default: []
+    required: false
+    type: list
+'''
+
+
+EXAMPLES = '''
+- name: Set or update some CI/CD variables
+  gitlab_project_variables:
+    server_url: https://gitlab.com
+    login_token: secret_access_token
+    name: markuman/dotfiles
+    purge_vars: False
+    vars:
+      - ACCESS_KEY_ID: abc123
```

Comment:
Since each of your attributes have only one attributes, what's the point of making a list ?
Usage of a simple dictionary should be more easier to understand.
```yaml
vars:
  ACCESS_KEY_ID: abc123
  SECRET_ACCESS_KEY: 321cba
```

Reply:
https://github.com/markuman/ansible/commit/52fa05633dae34520e99081918f93e35879818e4